src/shared/components/query/QueryStore.ts

```diff
@@ -1104,6 +1108,62 @@ export class QueryStore
     {
         return this.genesetIdsQuery.query;
     }
+    
+    readonly volcanoPlotTableData = remoteData<Geneset[]>({
+        invoke: async()=>{
+            const tableData: Geneset[] = [];
+            const hierarchyData = await getHierarchyData(
+                    this.getFilteredProfiles("GENESET_SCORE")[0].molecularProfileId, 
+                    Number(this.volcanoPlotSelectedPercentile.value),
+                    0, 1, this.defaultSelectedSampleListId);
+            for (const node of hierarchyData) {

```

Comment:
when you are writing logic which populates a variable (in this case tableData), you should try to use either map or reduce.  in this case, reduce would work because you are filtering.  the reason this is better is because the code is more semantic.  const result  = source.reduce(...).    a reader can see immediately that the aim of this code is to set the result variable to the return value of reduce.